src/ImageSharp/Formats/Heif/HeifDecoderCore.cs

```diff
@@ -0,0 +1,705 @@
+
+            // Skip Protection Index, not sure what that means...
+            bytesRead += 2;
+            Heif4CharCode itemType = (Heif4CharCode)BinaryPrimitives.ReadUInt32BigEndian(buffer[bytesRead..]);
+            bytesRead += 4;
+            item = new HeifItem(itemType, itemId);
+            item.Name = ReadNullTerminatedString(buffer[bytesRead..]);
+            bytesRead += item.Name.Length + 1;
+            if (item.Type == Heif4CharCode.Mime)
+            {
+                item.ContentType = ReadNullTerminatedString(buffer[bytesRead..]);
+                bytesRead += item.ContentType.Length + 1;
+
+                // Optional field.
+                if (bytesRead < boxLength)
+                {
+                    item.ContentEncoding = ReadNullTerminatedString(buffer[bytesRead..]);
+                    bytesRead += item.ContentEncoding.Length + 1;
+                }
+            }
+            else if (item.Type == Heif4CharCode.Uri)
+            {
+                item.UriType = ReadNullTerminatedString(buffer[bytesRead..]);
+                bytesRead += item.UriType.Length + 1;
+            }
+        }
+
+        if (item != null)
+        {
+            this.items.Add(item);
+        }
+
+        return bytesRead;
+    }
+
+    private void ParseItemReference(BufferedReadStream stream, long boxLength)
+    {
+        using IMemoryOwner<byte> boxMemory = this.ReadIntoBuffer(stream, boxLength);
+        Span<byte> boxBuffer = boxMemory.GetSpan();
+        int bytesRead = 0;
+        bool largeIds = boxBuffer[bytesRead] != 0;
+        bytesRead += 4;
+        while (bytesRead < boxLength)
+        {
+            bytesRead += ParseBoxHeader(boxBuffer[bytesRead..], out long subBoxLength, out Heif4CharCode linkType);
+            uint sourceId = ReadUInt16Or32(boxBuffer, largeIds, ref bytesRead);
+            HeifItemLink link = new(linkType, sourceId);
+
+            int count = BinaryPrimitives.ReadUInt16BigEndian(boxBuffer[bytesRead..]);
+            bytesRead += 2;
+            for (uint i = 0; i < count; i++)
+            {
+                uint destId = ReadUInt16Or32(boxBuffer, largeIds, ref bytesRead);
+                link.DestinationIds.Add(destId);
+            }
+
+            this.itemLinks!.Add(link);
+        }
+    }
+
+    private void ParsePrimaryItem(BufferedReadStream stream, long boxLength)
+    {
+        // BoxLength should be 6 or 8.
+        using IMemoryOwner<byte> boxMemory = this.ReadIntoBuffer(stream, boxLength);
+        Span<byte> boxBuffer = boxMemory.GetSpan();
+        byte version = boxBuffer[0];
+        int bytesRead = 4;
+        this.primaryItem = ReadUInt16Or32(boxBuffer, version != 0, ref bytesRead);
+    }
+
+    private void ParseItemProperties(BufferedReadStream stream, long boxLength)
+    {
+        // Cannot use Dictionary here, Properties can have multiple instances with the same key.
+        List<KeyValuePair<Heif4CharCode, object>> properties = new();
```

Comment:
@ynse01 shouldn't those properties be bound to an item? Otherwise they will be lost here after parsing or do I miss something?

Reply:
Sure they are connected, this is hanlded by the `ipma` box or `ParsePropertyAssociation()` in the code.